### PR TITLE
xtensa-build-zephyr.py: copy mtrace-reader.py to tools folder

### PR DESCRIPTION
**Pull request by keqiaozhang**
we will switch to using this new logging tool, copy it to the tools
folder, then it will be uploaded to CI storage after building.

Signed-off-by: Keqiao Zhang <keqiao.zhang@intel.com>

### CODE REVIEW

**Comment by keqiaozhang** [comment]
The Zephyr build failures are not caused by this PR. https://github.com/thesofproject/sof/pull/6258 can fix it.

**Review by kv2019i — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]
@marc-hb @fredoh9 good for you ?

File: scripts/xtensa-build-zephyr.py [hunk 1]
```diff
@@ -588,6 +588,12 @@ def build_platforms():
 			tools_output_dir,
 			symlinks=True, ignore_dangling_symlinks=True, dirs_exist_ok=True)
 
+	# mtrace
+	shutil.copytree(pathlib.Path(SOF_TOP) /
+			  "tools" / "mtrace",
```
Comment (marc-hb) [inline scripts/xtensa-build-zephyr.py:593]: Please append this file to `src_dest_list` instead, as recommended by the comment line 577. We already have two different code sections to install files, we don't need a third one.

**Review by marc-hb — APPROVED** [review]

**Review by marc-hb — CHANGES_REQUESTED** [review]
clicked too fast

File: scripts/xtensa-build-zephyr.py (hunk 1 above)
Comment (keqiaozhang) [reply scripts/xtensa-build-zephyr.py]: Updated.

**Review by marc-hb — APPROVED** [review]
Sorry I didn't realize you were copying an entire directory in earlier version 1ee70a28f9d86. I forgot `shutil.copytree` copies directories only. I only noticed the duplication with the previous line.

Now you're copying a single file, is that enough? If not enough and you really prefer to copy the entire `mtrace/` directory then feel free to add a `for` loop around the `shutil.copytree()`.

**Comment by keqiaozhang** [comment]
> you're copying a single file, is that enough?

I think it's enough ATM. @kv2019i , please help to double confirm it?

**Pull request merged**